pkg/controllers/deprovisioning/nodeclaim_consolidation_test.go

```diff
@@ -2264,6 +2267,327 @@ var _ = Describe("NodeClaim/Consolidation", func() {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				ExpectReconcileSucceeded(ctx, deprovisioningController, client.ObjectKey{})
+			}()
+
+			// Iterate in a loop until we get to the validation action
+			// Then, apply the pods to the cluster and bind them to the nodes
+			for {
+				time.Sleep(250 * time.Millisecond)
+				if fakeClock.HasWaiters() {
+					break
+				}
+			}
+			doNotEvictPods := test.Pods(2, test.PodOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						v1beta1.DoNotDisruptAnnotationKey: "true",
+					},
+				},
+			})
+			ExpectApplied(ctx, env.Client, doNotEvictPods[0], doNotEvictPods[1])
+			ExpectManualBinding(ctx, env.Client, doNotEvictPods[0], node)
+			ExpectManualBinding(ctx, env.Client, doNotEvictPods[1], node2)
+
+			// Step forward to satisfy the validation timeout and wait for the reconcile to finish
+			fakeClock.Step(45 * time.Second)
+			wg.Wait()
+
+			// we would normally be able to consolidate down to a single node, but we are blocked by the do-not-evict pods during validation
+			Expect(ExpectNodeClaims(ctx, env.Client)).To(HaveLen(2))
+			Expect(ExpectNodes(ctx, env.Client)).To(HaveLen(2))
+			ExpectExists(ctx, env.Client, node)
+			ExpectExists(ctx, env.Client, node2)
+		})
+		It("should not delete node if pods schedule with a blocking PDB during the TTL wait", func() {
+			pods := test.Pods(2, test.PodOptions{})
+			ExpectApplied(ctx, env.Client, nodePool, nodeClaim, node, nodeClaim2, node2, pods[0], pods[1])
+
+			// bind pods to node
+			ExpectManualBinding(ctx, env.Client, pods[0], node)
+			ExpectManualBinding(ctx, env.Client, pods[1], node2)
+
+			// inform cluster state about nodes and machines
+			ExpectMakeNodesAndNodeClaimsInitializedAndStateUpdated(ctx, env.Client, nodeStateController, nodeClaimStateController, []*v1.Node{node, node2}, []*v1beta1.NodeClaim{nodeClaim, nodeClaim2})
+
+			fakeClock.Step(10 * time.Minute)
+
+			var wg sync.WaitGroup
+
+			// Trigger the reconcile loop to start but don't trigger the verify action
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				ExpectReconcileSucceeded(ctx, deprovisioningController, client.ObjectKey{})
+			}()
+
+			// Iterate in a loop until we get to the validation action
+			// Then, apply the pods to the cluster and bind them to the nodes
+			for {
+				time.Sleep(250 * time.Millisecond)
+				if fakeClock.HasWaiters() {
+					break
+				}
+			}
+			labels := map[string]string{
+				"app": "test",
+			}
+			blockingPDBPods := test.Pods(2, test.PodOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+			})
+			pdb := test.PodDisruptionBudget(test.PDBOptions{
+				Labels:         labels,
+				MaxUnavailable: fromInt(0),
+				Status: &policyv1.PodDisruptionBudgetStatus{
+					ObservedGeneration: 1,
+					DisruptionsAllowed: 0,
+					CurrentHealthy:     1,
+					DesiredHealthy:     1,
+					ExpectedPods:       1,
+				},
```

Comment:
Do we need the PDB? In my experience, you just need to create one with the proper maxUnavailable, where this status will get updated by itself anyway.